app/components/attachment_button.js

```diff
@@ -35,6 +35,8 @@ export default class AttachmentButton extends PureComponent {
         onShowFileSizeWarning: PropTypes.func,
         theme: PropTypes.object.isRequired,
         uploadFiles: PropTypes.func.isRequired,
+        removeProfileImage: PropTypes.func,
```

Comment:
We also use this component in other places, so we should try to keep this behaviour separate from the component itself. What you could do is instead of adding props for `removeProfileImage` and `profileImageUri`, you could instead add a prop like `getExtraActions` so that the `EditProfile` screen can tell the `AttachmentButton` about the extra option to render.

Also, this part isn't really needed, but you could even take it one step further and make a separate component that wraps `AttachmentButton` to contain that logic and make the `EditProfile` screen a bit simpler. It could look like
```javascript
class ProfilePictureButton extends React.PureComponent {
    static propTypes = {
        currentUser: PropTypes.object.isRequired,
    };

    getExtraOptions = () => {
    	if (!this.isCustomProfileImage()) {
    		return null;
    	}

    	return [{
	    	action: this.removeProfileImage,
	    	text: 'Remove Image'
    	}];
    }

    render() {
	    const {children, ...props} = this.props;

	    // Avoid passing unneeded props
	    Reflect.deleteProperty(props, 'currentUser');

        return (
            <AttachmentButton
            	{...props}
            	getExtraOptions={this.getExtraOptions}
            >
            	{children}
            </AttachmentButton>
        );
    }
}
```

Reply:
I also wondered about that, as it is a global component. That looks like a good approach to keep it clean.